File: src/#852/solution.go
package solution

func peakIndexInMountainArray(A []int) int {
	peak := len(A) / 2
	for peak > 0 && peak < len(A) {
		if A[peak-1] < A[peak] && A[peak] > A[peak+1] {
			return peak
		}
		if A[peak-1] > A[peak] {
			peak--
		} else {
			peak++
		}
	}
	return peak
}

File: src/#852/solution.js
var peakIndexInMountainArray = function (A) {
  let peak = Math.floor(A.length / 2)

  while (peak > 0 && peak < A.length - 1) {
    if (A[peak - 1] < A[peak] && A[peak] > A[peak + 1]) {
      return peak
    }
    if (A[peak - 1] > A[peak]) {
      peak--
    } else {
      peak++
    }
  }
  return peak
}
